src/battle_controllers.c

```diff
@@ -0,0 +1,1551 @@
+}
+
+void EmitCmd34(u8 bufferId, u8 b, u8 *c)
+{
+    s32 i;
+
+    gBattleBuffersTransferData[0] = CONTROLLER_34;
+    gBattleBuffersTransferData[1] = b;
+    for (i = 0; i < 3; i++)
+        gBattleBuffersTransferData[2 + i] = c[i];
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 5);
+}
+
+void EmitCmd35(u8 bufferId, u16 b)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_35;
+    gBattleBuffersTransferData[1] = b;
+    gBattleBuffersTransferData[2] = (b & 0xFF00) >> 8;
+    gBattleBuffersTransferData[3] = 0;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd36(u8 bufferId, u16 b)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_36;
+    gBattleBuffersTransferData[1] = b;
+    gBattleBuffersTransferData[2] = (b & 0xFF00) >> 8;
+    gBattleBuffersTransferData[3] = 0;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd37(u8 bufferId)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_37;
+    gBattleBuffersTransferData[1] = 37;
+    gBattleBuffersTransferData[2] = 37;
+    gBattleBuffersTransferData[3] = 37;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd38(u8 bufferId, u8 b)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_38;
+    gBattleBuffersTransferData[1] = b;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 2);
+}
+
+void EmitCmd39(u8 bufferId)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_39;
+    gBattleBuffersTransferData[1] = 39;
+    gBattleBuffersTransferData[2] = 39;
+    gBattleBuffersTransferData[3] = 39;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd40(u8 bufferId)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_40;
+    gBattleBuffersTransferData[1] = 40;
+    gBattleBuffersTransferData[2] = 40;
+    gBattleBuffersTransferData[3] = 40;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitHitAnimation(u8 bufferId)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_HITANIMATION;
+    gBattleBuffersTransferData[1] = 41;
+    gBattleBuffersTransferData[2] = 41;
+    gBattleBuffersTransferData[3] = 41;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd42(u8 bufferId)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_42;
+    gBattleBuffersTransferData[1] = 42;
+    gBattleBuffersTransferData[2] = 42;
+    gBattleBuffersTransferData[3] = 42;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitEffectivenessSound(u8 bufferId, u16 songId)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_EFFECTIVENESSSOUND;
+    gBattleBuffersTransferData[1] = songId;
+    gBattleBuffersTransferData[2] = (songId & 0xFF00) >> 8;
+    gBattleBuffersTransferData[3] = 0;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitPlayFanfareOrBGM(u8 bufferId, u16 songId, bool8 playBGM)
```

Comment:
Shouldn't this be EmitPlayFanfareOrSE? Is this used for playing actual music or just sound effects and fanfares?

Reply:
No, it either play a fanfare or BGM